CMakeLists.txt

```diff
@@ -31,11 +31,6 @@ if (CMAKE_CXX_COMPILER_ID MATCHES Clang)
         "${CMAKE_CXX_FLAGS_RELEASE} -Wno-unused-parameter")
 endif()
 
-if ((NOT "${CMAKE_CXX_FLAGS_RELEASE}" MATCHES "std=c\\+\\+0x") AND
-    (NOT "${CMAKE_CXX_FLAGS_RELEASE}" MATCHES "std=c\\+\\+11"))
-    message(FATAL_ERROR "CXX flags not set properly. Remove CMakeCache.txt and run cmake again")
-endif()
-
```

Comment:
I added this because, many including me ran into this problem when trying to build SymEngine at first and realizing there's no CXX compiler installed. After installing the CXX compiler, you need to remove `CMakeCache.txt` for CXX flags to be set.

We could remove this and make the error message in `cmake/checkcxx11.cpp` more helpful